facade/src/main/java/org/jboss/pnc/facade/providers/DeliverableAnalyzerReportProviderImpl.java

```diff
@@ -0,0 +1,108 @@
+import org.jboss.pnc.model.DeliverableAnalyzerReport;
+import org.jboss.pnc.model.DeliverableArtifact;
+import org.jboss.pnc.spi.datastore.predicates.DeliverableArtifactPredicates;
+import org.jboss.pnc.spi.datastore.repositories.DeliverableAnalyzerReportRepository;
+import org.jboss.pnc.spi.datastore.repositories.DeliverableArtifactRepository;
+import org.jboss.pnc.spi.datastore.repositories.api.PageInfo;
+import org.jboss.pnc.spi.datastore.repositories.api.Predicate;
+import org.jboss.pnc.spi.datastore.repositories.api.SortInfo;
+
+import javax.annotation.security.PermitAll;
+import javax.ejb.Stateless;
+import javax.inject.Inject;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@PermitAll
+@Stateless
+public class DeliverableAnalyzerReportProviderImpl extends
+        AbstractProvider<Base32LongID, DeliverableAnalyzerReport, org.jboss.pnc.dto.DeliverableAnalyzerReport, org.jboss.pnc.dto.DeliverableAnalyzerReport>
+        implements DeliverableAnalyzerReportProvider {
+
+    private DeliverableAnalyzerReportRepository deliverableAnalyzerReportRepository;
+
+    private DeliverableArtifactRepository deliverableArtifactRepository;
+
+    private DeliverableAnalyzerReportMapper deliverableAnalyzerReportMapper;
+
+    private ArtifactMapper artifactMapper;
+
+    @Inject
+    public DeliverableAnalyzerReportProviderImpl(
+            DeliverableAnalyzerReportRepository repository,
+            DeliverableArtifactRepository deliverableArtifactRepository,
+            DeliverableAnalyzerReportMapper mapper,
+            ArtifactMapper artifactMapper) {
+        super(repository, mapper, DeliverableAnalyzerReport.class);
+
+        this.deliverableAnalyzerReportRepository = repository;
+        this.deliverableArtifactRepository = deliverableArtifactRepository;
+        this.deliverableAnalyzerReportMapper = mapper;
+        this.artifactMapper = artifactMapper;
+    }
+
+    @Override
+    public Page<AnalyzedArtifact> getAnalyzedArtifacts(
+            int pageIndex,
+            int pageSize,
+            String query,
+            String sort,
+            String id) {
+        Base32LongID entityId = mapper.getIdMapper().toEntity(id);
+        Predicate<DeliverableArtifact> rsqlPredicate = rsqlPredicateProducer
+                .getCriteriaPredicate(DeliverableArtifact.class, query);
+        PageInfo pageInfo = pageInfoProducer.getPageInfo(pageIndex, pageSize);
+        SortInfo<DeliverableArtifact> sortInfo = rsqlPredicateProducer.getSortInfo(DeliverableArtifact.class, sort);
+
+        List<AnalyzedArtifact> analyzedArtifacts = deliverableArtifactRepository
+                .queryWithPredicates(
+                        pageInfo,
+                        sortInfo,
+                        rsqlPredicate,
+                        DeliverableArtifactPredicates.withReportId(entityId))
+                .stream()
+                .map(this::deliverableArtifactToDto)
+                .collect(Collectors.toList());
```

Comment:
Basically all these lines could be abstracted by `AbstractProvider#queryForCollection`, but it's not possible, why?

In order to create `DeliverableArtifactProviderImpl`, which would extend `AbstractProvider`, we would need mapper, which maps between `AnalyzedArtifact` and `DeliverableArtifact`.
In this case would be enough just `EntityMapper<DeliverableArtifactPK, DeliverableArtifact, AnalyzedArtifact, AnalyzedArtifact>`. We're **unable** to create such mapper, since the `AnalyzedArtifact` DTO does not have any ID (since the REF (hence DTO) we use in this `EntityMapper` has to extend `DTOEntity`).

Note: The `AnalyzedArtifact` DTO has this structure (as written in https://docs.engineering.redhat.com/display/JP/Support+levels+for+Deliverable+Analysis):
{
    "builtFromSource": true,
    "brewId": number,
    "artifact": {ARTIFACT_DTO}
}

Hence we have got 2 options:
1) Let it this way without using `AbstractProvider#queryForCollection`
2) Use the `AbstractProvider#queryForCollection`, but then we need to create some reasonable 1:1 mapping between DeliverableArtifactPK and String (because of `DTOEntity`)